src/js/actions/history.js

```diff
@@ -153,6 +154,9 @@ define(function (require, exports) {
                                 count: count,
                                 selectedIndices: selectedIndices
                             });
+                        })
+                        .then(function () {
+                            return this.transfer(toolActions.resetBorderPolicies);
```

Comment:
Does this need to wait for the previous dispatch? Also, that previous dispatch doesn't need to be async.